ui/src/components/modals/DeleteModelVersionModal.js

```diff
@@ -0,0 +1,118 @@
+    {},
+    false
+  );
+
+  const servingEndpoint = version.endpoints.find(
+    (endpoint) => endpoint.status === "serving"
+  );
+
+  useEffect(() => {
+    if (isLoaded) {
+      setDeleteConfirmation("")
+      closeModal();
+      callback();
+    }
+  }, [isLoaded, closeModal, callback]);
+
+  const isActiveEndpoint = function(status){
+    return ["pending","running","serving"].includes(status)
+  }
+
+  useEffect(() => {
+    setActiveEndpoint(version.endpoints.filter(item => isActiveEndpoint(item.status)))
+    setInactiveEndpoint(version.endpoints.filter(item => item.status === "failed"))
+  }, [version])
+
+  return (
+    <EuiOverlayMask>
+      <EuiConfirmModal
+        title="Delete Model Version"
+        onCancel={closeModal}
+        onConfirm={undeployVersion}
+        cancelButtonText="Cancel"
+        confirmButtonText="Delete"
+        buttonColor="danger"
+        confirmButtonDisabled={deleteConfirmation !== `${model.name}-version-${version.id}` || activeEndpoint.length > 0}>
+        {isLoading && (
+          <EuiProgress
+            size="xs"
+            color="accent"
+            className="euiProgress-beforePre"
+          />
+        )}
+        {servingEndpoint ? (
+          <span>
+            You cannot delete this Model Version because there are <b> Model Endpoints</b> using this version. 
+            <br/> <br/> If you still wish to delete this model version, please <b>Stop Serving</b> this model version. <br/>
+          </span>          
+        ) : (
+          <div>
+            {activeEndpoint.length > 0 ? (
+              <span>
+                You cannot delete this Model Version because there are <b> {activeEndpoint.length} Endpoints</b> using this version. 
+                <br/> <br/> If you still wish to delete this model version, please <b>Undeploy</b> Endpoints that use this version. <br/>
+              </span>
+            ) : (
+              <div>
+                You are about to delete model <b>{model.name}</b> version <b>{version.id}</b>. This action cannot be undone. 
+
+                <br/> <br/> To confirm, please type "<b>{model.name}-version-{version.id}</b>" in the box below
+                  <EuiFieldText     
+                    fullWidth            
+                    placeholder={`${model.name}-version-${version.id}`}
+                    value={deleteConfirmation}
+                    onChange={(e) => setDeleteConfirmation(e.target.value)}
+                    isInvalid={deleteConfirmation !== `${model.name}-version-${version.id}`} />  
+              </div>
+            )}
+            <br></br>
+            {activeEndpoint.length === 0 && inactiveEndpoint.length > 0 && (
+                <p>Deleting this Model Version will also delete {inactiveEndpoint.length} <b>Failed</b> Endpoints using this version. </p>
+            )}
+
+            <EndpointsTableModelVersion
+              endpoints={activeEndpoint.length > 0 ? activeEndpoint : inactiveEndpoint}
```

Comment:
Can we move this `<EndpointsTableModelVersion />`  under the previous check for `{activeEndpoint.length > 0 ? (`, along with the modal's message? Having this here is a little bit misleading (makes it look like we will give preference to displaying active endpoints, but actually, if there are active endpoints, the message/action items on the user is also different).